src/core/public/theme/theme_service.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject, Subject } from 'rxjs';
+import { shareReplay, takeUntil } from 'rxjs/operators';
+import { InjectedMetadataSetup } from '../injected_metadata';
+import type { CoreTheme, ThemeServiceSetup, ThemeServiceStart } from './types';
+
+interface SetupDeps {
+  injectedMetadata: InjectedMetadataSetup;
+}
+
+export class ThemeService {
+  private theme$?: BehaviorSubject<CoreTheme>;
+  private stop$ = new Subject();
+
+  public setup({ injectedMetadata }: SetupDeps): ThemeServiceSetup {
+    const theme = injectedMetadata.getTheme();
+    this.theme$ = new BehaviorSubject<CoreTheme>({ darkMode: theme.darkMode });
```

Comment:
When are we going to support `theme` updates?

Reply:
When we'll eventually allow theme customization by the user. This would allow, for example, to preview the theme overrides dynamically. Having an observable for the theme is to anticipate this, and to avoid a breaking change in the API later when we'll have to support it.

But for now, I guess using `of({ darkMode: theme.darkMode })` instead of using a behavior subject would be perfectly fine, if that was the question

Reply:
> But for now, I guess using of({ darkMode: theme.darkMode }) instead of using a behavior subject would be perfectly fine, if that was the question

yes, let's start with more clear API.